src/services/mongoose/users.ts

```diff
@@ -29,6 +27,29 @@ export const UsersRepository = (): IUsersRepository => {
     }
   }
 
+  const findByIdAndUpdateLastConnectionDate = async (
```

Comment:
See explanation above - not appropriate for Repository.

Reply:
how to deal with that then? we don't want to retrieve this lastConnection to the local state because this is a write only operation so there is no need for prior state here. 

So the most efficient way to do it is just to update the value while we're getting the last state of the User.